packages/atomic/src/components/search/tabs/atomic-tab-section/atomic-tab-section.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  loadBreadcrumbActions,
+  loadQueryActions,
+  loadSearchActions,
+  loadSearchAnalyticsActions,
+  loadTabSetActions,
+} from '@coveo/headless';
+import {
+  Component,
+  h,
+  Element,
+  State,
+  Listen,
+  Prop,
+  EventEmitter,
+  Event,
+} from '@stencil/core';
+import ArrowDown from '../../../../images/arrow-down.svg';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {getActiveTab} from '../../../../utils/tab-utils';
+import {Button} from '../../../common/button';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-tab-section',
+  styleUrl: 'atomic-tab-section.pcss',
+  shadow: true,
+})
+export class AtomicTabSection {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element()
+  private host!: HTMLElement;
+
+  /**
+   * When provided, this tab will be selected by default when the component loads. Otherwise, the first tab is selected automatically.
+   */
+  @Prop() defaultActiveTab?: string;
+  /**
+   * Specifies whether the state should be cleared when the tab changes.
+   */
+  @Prop() clearStateOnTabChange?: boolean = false;
+
+  @State() public error!: Error;
+  @State() tabs: HTMLAtomicTabElement[] = [];
+
+  @Event({
+    eventName: 'atomic/tabInit',
+  })
+  tabInit!: EventEmitter;
+
+  @Listen('atomic/tabClick')
+  handleTabClick() {
+    this.updateActiveTab();
+    if (this.clearStateOnTabChange) {
+      this.clearState();
+    }
+  }
+
+  componentDidLoad() {
+    this.setInitialTab();
+  }
+
+  componentWillUpdate() {
+    const tabs = [...this.host.querySelectorAll('atomic-tab')];
+    this.tabs = tabs;
+
+    if (tabs.length === 0) {
+      this.error = new Error(
+        'The "atomic-tab-section" element requires at least one "atomic-tab" child.'
+      );
+      return;
+    }
+  }
+
+  setInitialTab() {
+    const initialTab =
+      this.tabs.find((tab) => tab.name === this.defaultActiveTab) ||
+      this.tabs[0];
+    const activeTab = getActiveTab(this.bindings.engine.state)?.tab;
+
+    if (initialTab && !activeTab) {
+      this.bindings.engine.dispatch(
+        loadTabSetActions(this.bindings.engine).updateActiveTab(initialTab.name)
+      );
+      this.tabInit.emit();
+    }
+    this.updateActiveTab();
+  }
+
+  buildDropdown() {
+    return this.tabs.map((tab) => (
+      <li>
+        <Button
+          style="text-transparent"
+          title={tab.name}
+          class="w-full px-6 pb-1 text-xl text-neutral-dark dropdown-option"
+          text={tab.label}
+          onClick={() => {
+            tab.select();
+          }}
+        />
+      </li>
+    ));
+  }
+
+  updateActiveTab() {
```

Comment:
All of this function should be removed, and we can simply know "what's the label of the current tab" by reading from Headless state.

Reply:
A different tab controller instance is created for each tab, so I would need to create a new tab manager controller and use that to get the active tab in the atomic-tab-manager component, right?

Reply:
I added a (very basic) tab-manager controller and refactored atomic-tab-section to use it